src/miniscript/types/extra_props.rs

```diff
@@ -494,6 +586,18 @@ impl Property for ExtData {
             ops_count_nsat: c
                 .ops_count_nsat
                 .and_then(|z| a.ops_count_nsat.map(|x| x + b.ops_count_static + z + 3)),
+            stack_elem_count_sat: cmp::max(
+                a.stack_elem_count_sat
+                    .and_then(|a| c.stack_elem_count_sat.map(|c| c + a)),
+                a.stack_elem_count_dissat
+                    .and_then(|a_dis| b.stack_elem_count_sat.map(|b| b + a_dis)),
+            ),
+            stack_elem_count_dissat: cmp::max(
+                a.stack_elem_count_sat
+                    .and_then(|a| c.stack_elem_count_dissat.map(|c| c + a)),
+                a.stack_elem_count_dissat
+                    .and_then(|a_dis| b.stack_elem_count_dissat.map(|b| b + a_dis)),
+            ),
```

Comment:
We should not add non-cannonical solutions here. We should remove first element in max because our signer never takes that path. Maybe if there is some demand for #56 (I doubt), we should keep things simple and only allow canonical options.  

See section malleability from http://bitcoin.sipa.be/miniscript/ for details for canonical vs non-cannonical if you are interested(not required for this PR).

Reply:
Definitely interested, will look into it.

Reply:
Ok, so i think your hash dissatisfaction below are a clear example of non-canonical dissatisfaction :p

Reply:
Those are not non-cannonical dissatisfactions. Non-cannonical ones are the ones require satisfaction of one fragment to dissatisfy another fragment. dissatisfaction hash check does not require any satisfaction, so it is not non-cannonical. 

But you are right, if any top level satisfaction of miniscript requires dissatisfaction of hashpreimages, then it is malleable. So, those can be safely changed to `None`.

I will bring up the issue about "How much effort we want to put into dealing with malleable scripts" with Andrew to also get his feedback. But that is an independent issue, and you are correct to change hash to None.